cpp/src/arrow/dataset/file_csv_test.cc

```diff
@@ -199,18 +210,11 @@ TEST_P(TestCsvFileFormat, OpenFailureWithRelevantError) {
   if (GetCompression() != Compression::type::UNCOMPRESSED) {
     GTEST_SKIP() << "File source name is different with compression";
```

Comment:
Is this really a problem?  Compression is just adding to the filename and we are testing with `HasSubstr`.